sbi/neural_nets/embedding_nets.py

```diff
@@ -266,19 +271,31 @@ def forward(self, x: Tensor) -> Tensor:
         Returns:
             Network output (batch_size, output_dim).
         """
-        batch, permutation_dim, _ = x.shape
 
-        iid_embeddings = self.trial_net(x.view(batch * permutation_dim, -1)).view(
-            batch, permutation_dim, -1
+        # Get number of trials from non-nan entries
+        trial_counts = x.shape[self.aggregation_dim] - torch.isnan(x[:, :, 0]).sum(
```

Comment:
will this work if `x` is an image?

Reply:
Probably not because we would need `x[:, :, 0, 0]`, no? Otherwise it sums over the very last dimension.

Reply:
good catch! fixed in the updated version.